legacy/build-deploy.sh

```diff
@@ -44,13 +44,27 @@ if [ ! -f .lagoon.yml ]; then
   echo "no .lagoon.yml file found"; exit 1;
 fi
 
+# Here we grab the git sha to populate build and runtime environment variables
+GIT_SHA=`git rev-parse HEAD`
 INJECT_GIT_SHA=$(cat .lagoon.yml | shyaml get-value environment_variables.git_sha false)
-if [ "$INJECT_GIT_SHA" == "true" ]
-then
-  LAGOON_GIT_SHA=`git rev-parse HEAD`
-else
-  LAGOON_GIT_SHA="0000000000000000000000000000000000000000"
-fi
+case "$INJECT_GIT_SHA" in
+  true|both)
+    LAGOON_GIT_SHA=$GIT_SHA
+    LAGOON_BUILD_GIT_SHA=$GIT_SHA
+    ;;
+  build)
+    LAGOON_BUILD_GIT_SHA=$GIT_SHA
+    LAGOON_GIT_SHA="0000000000000000000000000000000000000000"
+    ;;
+  runtime)
+    LAGOON_BUILD_GIT_SHA="0000000000000000000000000000000000000000"
+    LAGOON_GIT_SHA=$GIT_SHA
+    ;;
+  *)
+    LAGOON_GIT_SHA="0000000000000000000000000000000000000000"
+    LAGOON_BUILD_GIT_SHA=$LAGOON_GIT_SHA
+    ;;
+esac
```

Comment:
My only concerns with this are that the `environments.git_sha` field goes from being a boolean, to a string. I think it is ok though.